pkg/queue/readiness/probe_encoding.go

```diff
@@ -23,17 +23,27 @@ import (
 	corev1 "k8s.io/api/core/v1"
 )
 
-// DecodeProbe takes a json serialised *corev1.Probe and returns a Probe or an error.
-func DecodeProbe(jsonProbe string) (*corev1.Probe, error) {
-	p := &corev1.Probe{}
-	if err := json.Unmarshal([]byte(jsonProbe), p); err != nil {
-		return nil, err
+// DecodeProbes takes a json serialised *corev1.Probe OR []*corev1.Probe (depending on multiContainerProbes)
+// and returns a slice of probes or an error.
+func DecodeProbes(jsonProbe string, multiContainerProbes bool) ([]*corev1.Probe, error) {
+	probes := []*corev1.Probe{}
+	if multiContainerProbes {
```

Comment:
we also could just try to do both and drop the env variable? But it is a bit unnecessary as one case will always fail.